src/evo/netinfo.cpp

```diff
@@ -147,6 +147,29 @@ std::string NetInfoEntry::ToStringAddrPort() const
         m_data);
 }
 
+bool NetInfoInterface::IsEqual(const std::shared_ptr<NetInfoInterface>& lhs, const std::shared_ptr<NetInfoInterface>& rhs)
```

Comment:
why make this static and not operator== inside the class?

Reply:
see: https://coliru.stacked-crooked.com/a/0e68596a6becd1f6 Something like that may be useful. In general, I would prefer we find ways to do this without all the dynamic runtime type checking stuff

Reply:
The proposed solution is quite complex for the simple one parent (interface), two children (legacy and extended implementations) inheritance structure. A simpler one involving the much cheaper `typeid` operator has been used instead.

A helper function is still needed as `NetInfoInterface::IsEqual()` handled both downcasting and deep comparison and the solution mentioned only deals with the former but that can be replaced with a more generic helper function (`util::shared_ptr_{,not_}equal`), which we have done here.

Resolved in latest push.